File: app/src/main/java/org/mifos/mobilebanking/models/templates/loans/Timeline.java
package org.mifos.mobilebanking.models.templates.loans;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Rajan Maurya on 16/07/16.
 */
public class Timeline implements Parcelable {

    @SerializedName("expectedDisbursementDate")
    List<Integer> expectedDisbursementDate;

    public List<Integer> getExpectedDisbursementDate() {
        return expectedDisbursementDate;
    }

    public void setExpectedDisbursementDate(List<Integer> expectedDisbursementDate) {
        this.expectedDisbursementDate = expectedDisbursementDate;
    }

    @Override
    public String toString() {
        return "Timeline{" +
                "expectedDisbursementDate=" + expectedDisbursementDate +
                '}';
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeList(this.expectedDisbursementDate);
    }

    public Timeline() {
    }

    protected Timeline(Parcel in) {
        this.expectedDisbursementDate = new ArrayList<>();
        in.readList(this.expectedDisbursementDate, Integer.class.getClassLoader());
    }

    public static final Creator<Timeline> CREATOR = new Creator<Timeline>() {
        @Override
        public Timeline createFromParcel(Parcel source) {
            return new Timeline(source);
        }

        @Override
        public Timeline[] newArray(int size) {
            return new Timeline[size];
        }
    };
}

File: app/src/main/java/org/mifos/mobilebanking/models/accounts/savings/TimeLine.kt
package org.mifos.mobilebanking.models.accounts.savings

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import java.util.*

/**
 * Created by Rajan Maurya on 05/03/17.
 */

@Parcelize
data class TimeLine(
        @SerializedName("submittedOnDate")
        var submittedOnDate: List<Int> = ArrayList(),

        @SerializedName("submittedByUsername")
        var submittedByUsername: String,

        @SerializedName("submittedByFirstname")
        var submittedByFirstname: String,

        @SerializedName("submittedByLastname")
        var submittedByLastname: String,

        @SerializedName("approvedOnDate")
        var approvedOnDate: List<Int> = ArrayList(),

        @SerializedName("approvedByUsername")
        var approvedByUsername: String,

        @SerializedName("approvedByFirstname")
        var approvedByFirstname: String,

        @SerializedName("approvedByLastname")
        var approvedByLastname: String,

        @SerializedName("activatedOnDate")
        var activatedOnDate: List<Int>,

        @SerializedName("activatedByUsername")
        var activatedByUsername: String,

        @SerializedName("activatedByFirstname")
        var activatedByFirstname: String,

        @SerializedName("activatedByLastname")
        var activatedByLastname: String,

        @SerializedName("closedOnDate")
        var closedOnDate: List<Int>

) : Parcelable